mineral-ui/mineral-ui
# [Table] Render props

### Description

* Adds support for render props to Table component - `row`, `cell`, `headerCell`
* Adds a render props guidelines page
* Refactors all render props examples and API docs to point to render props guidelines page
* Refactors all render props tests to use snapshots.  Also updates Jest.
* Adds RenderCounter component to website to help debug render counts.  Not taking this any further right now as new React dev tools may obviate need for it.

### Motivation and context

closes #735 

### Screenshots, videos, or demo, if appropriate

https://735-table-render-props--mineral-ui.netlify.com/

### How to test

1. Verify the content in the [render props guidelines page](https://735-table-render-props--mineral-ui.netlify.com/render-props)
1. View the **"Custom"** render prop examples on the [Table component page](https://735-table-render-props--mineral-ui.netlify.com/components/table).
1. Test perf if you are brave.  Let's discuss.  Last WIP commit will be removed prior to merging.

### Types of changes

- New feature (non-breaking change which adds functionality)
  * Add render prop support to Table

- Breaking change (fix or feature that would cause existing functionality to change)
  * Rename theme variables that are used in sortable header cells. `TableSortableHeaderCell_border_focus`, `TableSortableHeaderCell_color_focus`

### Checklist
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change. - _Note that it won't work locally in CRA unless the last commit is removed.  It will be removed prior to merging._
